### PR TITLE
Assertions to exceptions

### PR DESCRIPTION
**Pull request by MocktaiLEngineer**
# What does this PR do?

This PR addresses the issue [#12789](url)
I have modified the file transformers/src/transformers/tokenization_utils.py file to throw a type error instead of assertions.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/tokenization_utils.py [hunk 1]
```diff
@@ -362,7 +362,8 @@ def _add_tokens(self, new_tokens: Union[List[str], List[AddedToken]], special_to
 
         tokens_to_add = []
         for token in new_tokens:
-            assert isinstance(token, str)
+            if not isinstance(token,str):
+                raise TypeError(f"Token ({token}) is not a string")
```
Comment (patil-suraj) [inline src/transformers/tokenization_utils.py:366]:
```suggestion
                raise TypeError(f"Token {token} is not a string.")
```

**Review by patil-suraj — APPROVED** [review]
Thank you for fixing this!

Could you run `make style` and `make quality` from the root of the repo? That should fix the failing test.

File: src/transformers/tokenization_utils.py [hunk 2]
```diff
@@ -362,7 +362,8 @@ def _add_tokens(self, new_tokens: Union[List[str], List[AddedToken]], special_to
 
         tokens_to_add = []
         for token in new_tokens:
-            assert isinstance(token, str)
+            if not isinstance(token,str):
+                raise TypeError(f"Token {token} is not a string.")
```
Comment (sgugger) [inline src/transformers/tokenization_utils.py:366]:
```suggestion
                raise TypeError(f"Token {token} is not a string but a {type(token)}.")
```

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

**Comment by sgugger** [comment]
Note that you still need to run `make style` on your branch to fix the code quality check.

**Comment by MocktaiLEngineer** [comment]
> Thank you for fixing this!
> 
> Could you run `make style` and `make quality` from the root of the repo? That should fix the failing test.

Thank you so much, will do!

**Comment by MocktaiLEngineer** [comment]
> Note that you still need to run `make style` on your branch to fix the code quality check.

Yes, this is my first time contributing to a PR, so I'm sorry if I'm being a little slow. I have run the 'make style', hoping it passes this time. And thanks for approving the changes :)

**Comment by sgugger** [comment]
No problem, we all have to learn and start somewhere :-)
The last test failure is unrelated to this PR (flaky test) so merging. Thanks again for your contribution!

**Pull request merged**